docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,107 @@
+---
+id: slot-validation-actions
+sidebar_label: Slot Validation Actions
+title: Slot Validation Actions
+abstract: A slot validation action is a special type of custom action, designed to handle custom extraction and/or validation of slot values. This can be used to validate slots with predefined mappings or extract slots with custom mappings.
+---
+
+Rasa SDK provides three base classes that you can inherit from in order to implement custom slot extraction and
+validation logic:
+
+## `Action` class
+
+You can use this class to create a regular [custom action](./custom-actions.mdx). The base class will not prevent you
+from returning arbitrary events, however a slot validation action should only return `SlotSet` and `BotUttered` events.
+Any other event type will be filtered out by the default action [`action_extract_slots`](./default-actions.mdx#action_extract_slots).
+The name of the custom action must be specified in the `action` key of the relevant
+[custom slot mapping](./domain.mdx#custom-slot-mappings) in the domain.
+Note that the action name must be listed in the domain `actions` section too.
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` class in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+
+:::note
+`ValidationAction` will ignore any slots mapped to a form added to the slot mapping's `conditions`.
+Please extend the [`FormValidationAction`](./slot-validation-actions.mdx#formvalidationaction-class) class for applying
+custom slot mappings only within the context of a form.
+:::
```

Comment:
```suggestion
:::note
`ValidationAction` is intended for extracting slots **outside**  the context of a form. It will ignore extraction and validation methods for any slots that have a form specified under the slot mapping's `conditions`. It will not run these methods when the specified form is active nor when no form is active.
Please extend the [`FormValidationAction`](./slot-validation-actions.mdx#formvalidationaction-class) class for applying
custom slot mappings only within the context of a form.
:::
```

Reply:
Whoop whoop so excited about merging this, yipeee 😄